src/explore-education-statistics-common/src/modules/table-tool/components/LocationFiltersForm.tsx

```diff
@@ -129,42 +131,47 @@ const LocationFiltersForm = ({
     };
   }, [initialValues, hasSingleOption, keyedOptions, options]);
 
+  const validationSchema = useMemo<ObjectSchema<LocationFormValues>>(() => {
+    return Yup.object({
+      locations: Yup.object().test(
+        'required',
+        'Select at least one location',
+        (value: Dictionary<string[]>) =>
+          Object.values(value).some(groupOptions => groupOptions.length > 0),
+      ),
+    });
+  }, []);
+
+  const handleSubmitForm = async (values: LocationFormValues) => {
+    const locationIds = Object.values(values.locations).flat();
+    await goToNextStep(async () => {
+      await onSubmit({ locationIds });
+    });
+  };
+
   return (
-    <Formik<LocationFormValues>
+    <FormProvider
       enableReinitialize
       initialValues={initialFormValues}
-      validateOnBlur={false}
-      validationSchema={Yup.object<LocationFormValues>({
-        locations: Yup.object().test(
-          'required',
-          'Select at least one location',
-          (value: Dictionary<string[]>) => {
-            return Object.values(value).some(
-              groupOptions => groupOptions.length > 0,
-            );
-          },
-        ),
-      })}
-      onSubmit={async values => {
-        const locationIds = Object.values(values.locations).flat();
-        await goToNextStep(async () => {
-          await onSubmit({ locationIds });
-        });
-      }}
+      validationSchema={validationSchema}
     >
-      {form => {
+      {({ formState, getValues, reset }) => {
+        const showError =
+          (!formState.isValid && Object.keys(formState.touchedFields).length) ||
+          (!formState.isValid && formState.isSubmitted);
         if (isActive) {
           return (
-            <Form {...form} id={formId} showSubmitError>
+            <RHFForm
+              id={formId}
```

Comment:
We can inline the `formId` variable nowadays as the ids get autogenerated.

Same applies for any other form in this PR.